### DISCUSSION TITLE
Deck estimation

### DISCUSSION THREAD

**Discussion by Maleware** [svidgen/mtg.deckanalyzer.com-Feedback#44] [General]
Hey all, I am coding my own analyzer but specific for EDH. I got stuck here and there, wondering how to determine the state of a deck "working". 

So a few technicalities first:
1. How did you order the structure of a deck? 
2. Did you do some "higher level" categorization of cards (Like draw, board wipe, wincon, etc.) If so, how did you manage to separate the cards in those piles?
3. Can you detect combos? If so, how are you evaluating those in terms of deck power?

For my thing, I am up building an archetype detector, which I would love to have some ideas and help.

Thanks for your replies in advance,
Maxi

**Comment by svidgen** [comment]
> How did you order the structure of a deck?

I'm not sure what this question is trying to get at. Are you trying to figure out how the analysis algorithms work?

> Did you do some "higher level" categorization of cards (Like draw, board wipe, wincon, etc.) If so, how did you manage to separate the cards in those piles?

Cards are tagged _by the user_ into these categories for the purpose of our core percentage score. We do perform separate, rough analysis for mana stats just based on common card attributes, like cmc and lands. Neither level of analysis accounts for mana acceleration, card draw, or card fetching at this time. (But it's on the radar.)

> Can you detect combos? If so, how are you evaluating those in terms of deck power?

I have two main concepts of "power" in the analyzer.

First is just adherence to user-defined tactics. That's the `P(All)` number. It basically says, how often will I draw the card categories I expect to draw *as I have defined them*.

Because each card is user tagged into "tactics", and each tactic has success criteria, we can just perform monte carlo simulations. (I.e., shuffle a lot of times and see what happens.) Each tactic is tested against each shuffling for success. If all tactics succeed against a shuffling, the shuffling is counted as a successful instance. And then we just compute a percentage.

There's a small margin of error there. The tactic-by-tactic percentages are deterministic aggregates of hypergeometric distributions. If you're unfamiliar, just google it and start studying!

The other concept of "power" is based on a really hacky ML I wrote and ran off to the side, based on some hand-ranked cards. The output of that ML is a function that scores cards based on attributes like CMC, power, toughness, etc.. The concept is neat, but it's very very beta; haven't taken this to the next level of soliciting user card ranking and re-running the ML regularly.

That notion of power is currently only used as a secondary sort field in the pruning suggestions.

> For my thing, I am up building an archetype detector, which I would love to have some ideas and help.

That's a level of analysis we haven't gotten to yet -- taking a deck, or even looking at the body of decks submitted and trying to derived meaning card combinations, for example. 

It's in the radar though.

Hope that gives the answers you need. Good luck!

--

PS. PTAL https://github.com/svidgen/mtg.deckanalyzer.com-Feedback/discussions/45
